coral-trino/src/test/java/com/linkedin/coral/trino/rel2trino/HiveToTrinoConverterTest.java

```diff
@@ -138,6 +138,19 @@ public void testViews(String database, String view, String expectedSql) {
             + "\"if\"(\"REGEXP_LIKE\"('rocks', '^[^\\\"]*$'), CAST(\"json_extract\"(\"$cor0\".\"b\".\"b1\", '$[\"' || 'rocks' || '\"]') AS VARCHAR(65535)), NULL) AS \"f\"\n"
             + "FROM (VALUES  (0)) AS \"t\" (\"ZERO\")) AS \"t0\"" },
 
-        { "test", "get_json_object_view", "SELECT \"json_extract\"(\"b\".\"b1\", '$.name')\nFROM \"test\".\"tablea\"" } };
+        { "test", "get_json_object_view", "SELECT \"json_extract\"(\"b\".\"b1\", '$.name')\nFROM \"test\".\"tablea\"" },
+
+        { "test", "view_from_utc_timestamp", "SELECT "
+            + "CAST(\"at_timezone\"(\"from_unixtime_nanos\"(CAST(\"a_tinyint\" AS BIGINT) * 1000000), 'America/Los_Angeles') AS TIMESTAMP), "
+            + "CAST(\"at_timezone\"(\"from_unixtime_nanos\"(CAST(\"a_smallint\" AS BIGINT) * 1000000), 'America/Los_Angeles') AS TIMESTAMP), "
+            + "CAST(\"at_timezone\"(\"from_unixtime_nanos\"(CAST(\"a_integer\" AS BIGINT) * 1000000), 'America/Los_Angeles') AS TIMESTAMP), "
+            + "CAST(\"at_timezone\"(\"from_unixtime_nanos\"(CAST(\"a_bigint\" AS BIGINT) * 1000000), 'America/Los_Angeles') AS TIMESTAMP), "
```

Comment:
Did you consider using `from_unixtime` (without  `_nanos`), to stay with lower precision timestamp where possible?

Reply:
Makes sense. I think I itiially though that from_unixtime only allows for second precision.
I updated the case for fractional types to use `from_unixtime`. For bigint/integer/smallint/tinyint I left `from_unixtime_nanos` to avoid unnecessary to_double conversion and division.